src/coreclr/vm/olevariant.cpp

```diff
@@ -329,20 +329,24 @@ VARTYPE OleVariant::GetVarTypeForTypeHandle(TypeHandle type)
 #endif
 
 #ifdef FEATURE_COMINTEROP
-    if (CoreLibBinder::IsClass(pMT, CLASS__DISPATCH_WRAPPER))
-        return VT_DISPATCH;
-    if (CoreLibBinder::IsClass(pMT, CLASS__UNKNOWN_WRAPPER))
-        return VT_UNKNOWN;
-    if (CoreLibBinder::IsClass(pMT, CLASS__ERROR_WRAPPER))
-        return VT_ERROR;
-    if (CoreLibBinder::IsClass(pMT, CLASS__CURRENCY_WRAPPER))
-        return VT_CY;
-    if (CoreLibBinder::IsClass(pMT, CLASS__BSTR_WRAPPER))
-        return VT_BSTR;
+    // The wrapper types are only available when built-in COM is supported.
+    if (EEConfig::IsBuiltInCOMSupported())
+    {
+        if (CoreLibBinder::IsClass(pMT, CLASS__DISPATCH_WRAPPER))
+            return VT_DISPATCH;
+        if (CoreLibBinder::IsClass(pMT, CLASS__UNKNOWN_WRAPPER))
+            return VT_UNKNOWN;
+        if (CoreLibBinder::IsClass(pMT, CLASS__ERROR_WRAPPER))
+            return VT_ERROR;
+        if (CoreLibBinder::IsClass(pMT, CLASS__CURRENCY_WRAPPER))
+            return VT_CY;
+        if (CoreLibBinder::IsClass(pMT, CLASS__BSTR_WRAPPER))
+            return VT_BSTR;
 
-    // VariantWrappers cannot be stored in VARIANT's.
-    if (CoreLibBinder::IsClass(pMT, CLASS__VARIANT_WRAPPER))
-        COMPlusThrow(kArgumentException, IDS_EE_COM_UNSUPPORTED_SIG);
+        // VariantWrappers cannot be stored in VARIANT's.
+        if (CoreLibBinder::IsClass(pMT, CLASS__VARIANT_WRAPPER))
+            COMPlusThrow(kArgumentException, IDS_EE_COM_UNSUPPORTED_SIG);
+    }
 #endif // FEATURE_COMINTEROP
 
```

Comment:
This is a more surgical fix than I was thinking and thanks for doing this!

I assume the code under the other #FEATURE_COMINTEROP in this method will not cause a problem?

Reply:
The other ones won't cause problems.

We'll always have SafeHandle and CriticalHandle (which arrays of are always unsupported in interop).

And then the rest is based on RCW/CCW support having been used to create a wrapper, which is already blocked previously.